### PR TITLE
Set up associations

### PR DESCRIPTION
**Pull request by ashmalzahra**
- Create a class `Classroom` with the following:
  - `@label` instance variable, should be initialized in the constructor.
  - Setter and getter for `@label` (remember about `attr_accessor`).
- Create the `has-many`/`belongs-to` relationship between `Classroom` and `Student`. The following should be implemented:
  - Create the `has-many` side (a classroom has many students).
  - Create the `belongs-to` side (a student belongs to a classroom).
  - Make sure that when adding a student to a classroom it also sets the classroom for the student.
  - Make sure that when setting the classroom for a student it also adds it to the classrooms' students.
- Create a class `Book` with the following:
  - `@title` and `@author` instance variables, should be initialized in the constructor.
  - Setters and getters for instance variables (remember about `attr_accessor`).
- Create a class `Rental` with the following:
  - `@date` instance variable, should be initialized in the constructor.
  - Setter and getter for `@date` (remember about `attr_accessor`).
- Create the `many-to-many` (also `has-many-through`) relationship between `Person` and `Book` using the intermediate class `Rental`. The following should be implemented:
  - Create the `has-many` side of `Book` and `Rental` (a book has many rentals).
  - Create the `belongs-to` side of `Rental` and `Book` (a rental belongs to a book).
  - Create the `has-many` side of `Person` and `Rental` (a person has many rentals).
  - Create the `belongs-to` side of `Rental` and `Person` (a rental belongs to a person).
  - Modify the constructor of `Rental` so `Book` and `Person` are set in it.

### CODE REVIEW

File: book.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+class Book
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rental = []
+  end
+
+  attr_accessor :title, :author
+  attr_reader :rental
```
Comment (rloterh) [inline book.rb]: - [ ] `rental` should be marked with `attr_accessor`, as we intend to both _access (get)_ as well as _write (set)_ it. Kindly ensure to improve upon this aspect.

File: book.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+class Book
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rental = []
+  end
+
```
Comment (rloterh) [inline book.rb:7]: - [x] A _rental_ can have many _books_, and _books_ can be subjected to many _rentals_. In order for us to establish a _many-to-many_ association between _rentals_ and _books_ (let's bear in mind that the association between _person_ and _rentals_ is also _many-to-many_), let's define a method for adding rentals within the book object like below:

```ruby
class Book
  attr_accessor :title, :author, :rentals
  .
  .

  def add_rental(person, date)
    Rental.new(date, self, person)
  end
end
```

File: rental.rb [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+
+    @person = person
+    person.rental << self
+
+    @book = book
+    book.rental << self
+  end
+
+  attr_accessor :date
+  attr_reader :person, :book
```
Comment (rloterh) [inline rental.rb]: - [x] Similarly, `book` and `person` should be marked with `attr_accessor`, as we intend to both _access (get)_ as well as _write (set)_ to it. Kindly ensure to improve upon this aspect.

File: person.rb [hunk 4]
```diff
@@ -1,4 +1,7 @@
 class Nameable
+  attr_accessor :name, :age
+  attr_reader :id, :rental
```
Comment (rloterh) [inline person.rb]: - [x] Again, we intend to both _access (get)_ as well as _write (set)_`rental`, and therefore should be marked with `attr_accessor`. Kindly ensure to improve upon this aspect.

Comment (rloterh) [inline person.rb]: - [x] Let's ensure a method for adding _rentals_ is present, so as to establish a a _rental_ `belong-to` _person_ relationship, like the code example below:

```ruby
class Person
  attr_accessor :name, :age, :rentals
  .
  .

  def add_rental(book, date)
    Rental.new(date, book, self)
  end

  .
  .
end
```

**Review by rloterh — CHANGES_REQUESTED** [review]
Hi Ashmal, 

You did extremely great on this project! 👏 👏 
There are however a few issues that you still need to work on to go to the next project but you are almost there! 
🏁  🌴 🚦 🚗 🏫  🏡   _🏃 

## Highlights 👍  
- Good job so far establishing associations for a book library objects! 
- ReadMe looks good!

## Required Changes ♻️
_Kindly check the inline comments under the review._

Cheers and Happy coding!👏 💻 🍷

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Ismailco — APPROVED** [review]
### Hi @ashmalzahra, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

cr-approve


## Highlights

✔️ No linter errors
✔️ Used Gitflow
✔️ Work documented professionally

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**